### ISSUE TITLE
Select2 $digest already in progress when model is updated

### ISSUE THREAD

**Issue by gduquesnay** [angular-ui/angular-ui-OLDREPO#315]
Trying to set the model that is bound to a select2 throws $digest already in progress.

click on test in the jsfiddle bellow and see the error in the console.
http://jsfiddle.net/7AvKZ/38/

Occurs with the master but doesn't seem to happen with version select2-3.2
http://plnkr.co/edit/gOf4nWzpmjg73GLMZI0m?p=preview

**Comment by ProLoser** [comment]
1. Really helps to NOT used the minified version during debugging / development
2. You should also mention that you're getting this error with AngularJS v1.1.1 (is this stable?)
3. Copying the master code into the plunker seems to _not_ have errors either
4. I only have 1 $scope.$apply() in the entire Select2 directive, which happens inside of `elm.bind('change')`
   - Does this mean the new API automatically does `$scope.$apply()` on `elm.bind()`?
   - Why would programmatically setting the value trigger a change event?

I will try to look into this further as soon as i find some time, but seeing as how things have been getting pretty backed up hopefully this information may prove useful for you to help us out (which would be **greatly appreciated**) and do some debugging of your own and may even result in a super-awesome PullRequest with the fix!

**Comment by tiagoroldao** [comment]
Angular UI 3.2 + Angular 1.0.4 seems to solve this issue. I have been ignoring it while working on other aspects of my project (seeing as I wasn't able to do anything about it), postponing solving it, and just realized it to be solved.

(working with Select2 - 3.2)

**Comment by ProLoser** [comment]
@gduquesnay can you confirm?

**Comment by ProLoser** [comment]
@joseym if you have some time can you take @gduquesnay's fiddle and confirm that the changes mentioned by @tiagoroldao fixes it so we can close this bug?

**Comment by joseym** [comment]
On it.

**Comment by joseym** [comment]
@ProLoser, @tiagoraldao suggestion doesn't seem to effect the $digest error: http://jsfiddle.net/joseym/FvVvF/

`angular 1.0.4` using `angular-ui 0.3.2`.

isn't 1.1.1 stable?

**Comment by ProLoser** [comment]
You guys really shouldn't be trying to debug stuff with minified versions.

Does the v0.4.0 select2 code fix his issue?

**Comment by joseym** [comment]
I also tried with unmini - looking into 0.4.0 now, where can I find/build that?

**Comment by tiagoroldao** [comment]
It does work - see this fiddle: http://jsfiddle.net/cGTdh/1/
I only changed the select2 source (noticed it was a pre-build source), and it all fell into place. I also changed the jQuery version, but upon retesting, it has no effect.
I don't believe the original error to be due to using an unbuilt version, though (as I said, I've been having the same problem, with the _right_ build, up until using 1.0.4)

@joseym the 1.1.x versions of angular are considered unstable. 1.1.1 is basically 1.0.3 with added (experimental) changes, and as 1.0.4 was released, so was 1.1.2. when the 1.1.x changes become stable, you get 1.2 :)
